pkg/uploader/provider/kopia.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright The Velero Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/kopia/kopia/snapshot/snapshotfs"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/vmware-tanzu/velero/pkg/uploader"
+	"github.com/vmware-tanzu/velero/pkg/uploader/kopia"
+
+	"github.com/vmware-tanzu/velero/internal/credentials"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	repokeys "github.com/vmware-tanzu/velero/pkg/repository/keys"
+	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo"
+	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo/service"
+)
+
+//BackupFunc mainly used to make testing more convenient
+var BackupFunc = kopia.Backup
+var RestoreFunc = kopia.Restore
+
+//kopiaProvider recorded info related with kopiaProvider
+type kopiaProvider struct {
+	bkRepo        udmrepo.BackupRepo
+	credGetter    *credentials.CredentialGetter
+	uploader      *snapshotfs.Uploader
+	restoreCancel chan struct{}
```

Comment:
Is it only for one specific restore? If it is, it should be not the field of `kopiaProvider` as `kopiaProvider` are shared by different `RunRestore()` actions

Reply:
I've removed it from the `kopiaProvider`  struct.